.pre-commit-config.yaml

```diff
@@ -1,22 +1,25 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 repos:
-    - repo: https://github.com/charliermarsh/ruff-pre-commit
-        # Ruff version.
-      rev: 'v0.0.285'
+    - repo: local
       hooks:
-          - id: ruff
-            args: [ --fix, --exit-non-zero-on-fix ]
+        - id: black
+          name: Format with Black
+          entry: poetry run black
+          language: system
+          types: [python]
 
-    - repo: https://github.com/ambv/black
-      rev: 23.3.0
-      hooks:
-          - id: black
-            language_version: python3
+        - id: ruff
+          name: Format with Ruff
+          entry: poetry run ruff
+          language: system
+          types: [ python ]
+          args: [ --fix, --exit-non-zero-on-fix, . ]
 
-    - repo: https://github.com/pre-commit/mirrors-mypy
-      rev: 'v1.3.0'  # Use the sha / tag you want to point at
-      hooks:
         - id: mypy
-          args: [--install-types, --non-interactive]
```

Comment:
Why remove `--install-types` ?

Reply:
I moved all mypy config to poetry.yaml so no need to add additional args here